torchao/prototype/float8nocompile/float8nocompile_linear_test.py

```diff
@@ -0,0 +1,90 @@
+import pytest
+import torch
+
+from torchao.float8.config import Float8LinearConfig
+from torchao.float8.float8_linear import manual_float8_matmul_with_args_in_hp
+from torchao.float8.float8_tensor import LinearMMConfig, ScaledMMConfig
+from torchao.prototype.float8nocompile.float8nocompile_linear import (
+    matmul_with_args_in_hp,
+)
+from torchao.prototype.float8nocompile.float8nocompile_scaling_utils import (
+    KernelAlgorithm,
+)
+
+
+# unit test comparing the two implementations
+@pytest.mark.parametrize(
+    "input_shape",
+    [(32, 16), (1, 32, 16), (2, 32, 16)],
+)
+def test_matmul_with_args_in_hp(input_shape: tuple[int, int]):
+    assert torch.cuda.is_available()
+    device = "cuda"
+
+    # high precision inputs
+    input_bf16 = torch.randn(
+        input_shape, dtype=torch.bfloat16, device=device, requires_grad=True
+    )
+    x_input_bf16 = input_bf16.clone().detach().to(device).requires_grad_(True)
```

Comment:
nit: the `x` and `y` var names are a bit confusing, as `y = torch.mm(x, w)` is widely used in a similar context

Reply:
makes sense, updated vars prefixed with `x_` to prefix with `prod_`, and updated vars prefixed with `y_` to prefixed with `prototype_`